scripts/lib/semaphore.py

```
@@ -66,16 +94,37 @@ def api_request(
-        try:
-            with urlopen(request, timeout=30) as response:
-                response_data = response.read().decode("utf-8")
-                return json.loads(response_data) if response_data else {}
-        except HTTPError as e:
-            error_body = e.read().decode("utf-8") if e.fp else ""
-            raise DeploymentError(f"API error {e.code}: {error_body}")
-        except URLError as e:
-            raise DeploymentError(f"Connection error: {e.reason}")
+        for attempt in range(1, attempts + 1):
+            try:
+                with urlopen(request, timeout=30) as response:
```

Ruff (0.15.17)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)